### ISSUE TITLE
duplicate key on cache_data

### ISSUE THREAD

**Issue by sysmonk** [grafana/grafana#17079]
**What happened**:
We've upgraded grafana from 6.1.6 to 6.2.0-beta1 recently due to the need to be able to query ElasticSearch 7 cluster. After the upgrade, sometimes an error pop's up in grafana dashboard for different requests - annotation, msearch result in 500. After a refresh, everything would continue to work. 
The error is: Error 1062: Duplicate entry "auth-proxy-sync-ttl:sysmonk" for key "PRIMARY"
This is a MySQL error. After debugging a bit, it looks to be cause by [this code](https://github.com/grafana/grafana/blob/2fff8f77dcdc90ab9a4890eeed95d8f3dced370b/pkg/infra/remotecache/database_storage.go#L107) which was added in #15457
There's probably a race condition happening, and the insert happens when there's no data yet, but data shows up during the request. 
My best suggestions is to use INSERT ON DUPLICATE KEY UPDATE, so something like this:
```
sql := `INSERT INTO cache_data (cache_key,data,created_at,expires) VALUES(?,?,?,?)`
		_, err = session.Exec(sql, key, data, getTime().Unix(), expiresInSeconds)
```
to:
```
sql := `INSERT INTO cache_data (cache_key,data,created_at,expires) VALUES(?,?,?,?) ON DUPLICATE KEY UPDATE data = ?, created_at = ?, expires = ?`
		_, err = session.Exec(sql, key, data, getTime().Unix(), expiresInSeconds, data, getTime().Unix(), expiresInSeconds)
```

I haven't looked at what other databases are supported, but this would work on mysql/mariadb. It's not a default SQL statement, so might not work on other sql databases - in example, sqlite has a [different syntax](https://www.sqlite.org/lang_UPSERT.html) for this.

**What you expected to happen**:
For stuff to just work, without errors :)

**How to reproduce it (as minimally and precisely as possible)**:
Enable sql backend
Enable proxy
Dashboard with more than one view
Doesn't happen straight away, in our case happens a few times per hour only.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.2.0-beta1
- Data source type & version: influxdb, elasticsearch 7
- OS Grafana is installed on: Debian Linux
- User OS & Browser: OSX 10.14.4 / Chrome 73
- Grafana plugins: 
- Others:

**Comment by bergquist** [comment]
Thank you for reporting this. Fix incoming.

**Comment by Jean-Daniel** [comment]
Is this issue supposed to be fixed in 6.2 final ? I'm actually running grafana 6.2 (`Version 6.2.0 (commit: 5d16da7, branch: HEAD)`) and have this error from time to time.

`Error 1062: Duplicate entry 'auth-proxy-sync-ttl:****' for key 'PRIMARY'`
